### DISCUSSION TITLE
Some questions about updating the dotfiles.

### DISCUSSION THREAD

**Discussion by FurtherSecrets24680** [mylinuxforwork/dotfiles#70] [Q&A]
Hey! I have been using this dotfiles for some months now. Aside some minor bugs, I have been greatly satisfied with the looks and features. Before, I installed this dotfiles from GitLab. Then, when a new update came out, I could install it directly from the ML4W dotfiles app. 

I got to know that the project moved to GitHub. Since then, a new version **2.9.3** got released. Now I have some questions:

1. Why are there no update options in the dotfiles app? Do I have to install it with the script again?
2. If I do install with the script, will all of my configs get overwritten?
3. I have customized the .bashrc , keybinds, autostart, fastfetch etc. configs to fit my needs. Will it be there? Or do I have to make a backup?

**Comment by mylinuxforwork** [comment]
Hi.

thanks for your questions.

You can update your dotfiles just by running the script again.

```
bash <(curl -s https://raw.githubusercontent.com/mylinuxforwork/dotfiles/main/setup.sh)
```

The script will detect that you have already an older version installed and will offer some restoring functions.

The script will offer to create a backup from your existing dotfiles in dotfiles-versions/backup. You can restore your modifications afterwards.

The .bashrc can be restored with the installation script directly.

After installing the new version you will get more options to protect your customizations with the updated hook.sh:
https://github.com/mylinuxforwork/dotfiles/wiki/Hook-and-Post-Installation-Scripts

And for your Hyprland configurations I suggest to create custom variations:
https://github.com/mylinuxforwork/dotfiles/wiki/Configuration-Variations

**Reply by FurtherSecrets24680** [reply]
Thanks for the answer!

**Comment by Veicm** [comment]
Now you can also start the Update with you're favourite aur helper.
If you use yay:
```
yay -S ml4w-hyprland
ml4w-hyprland-setup
```
Or for the rolling reales:
```
yay -S ml4w-hyprland-git
ml4w-hyprland-setup
```
